### PR TITLE
Remove Client#ClusterAvailable from interface

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
For Client#ClusterAvailable, since we don't care about the map for StatusResponse, we can return upon going through every endpoint.

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cmd/kubeadm/app/util/etcd/etcd.go
Comment (neolit123) [inline cmd/kubeadm/app/util/etcd/etcd.go]: i don't think this is correct. by using GetClusterStatus, the original code goes trough all the endpoints and checks for errors even if it ignores the actual responses.

/assign @fabriziopandini

Comment (tedyu) [reply cmd/kubeadm/app/util/etcd/etcd.go]: @neolit123 
Your comment has been addressed.

File: cmd/kubeadm/app/util/etcd/etcd.go
Comment (neolit123) [inline cmd/kubeadm/app/util/etcd/etcd.go]: but at this point this serves as code duplication.
i will defer to the opinion of @fabriziopandini here.

Comment (tedyu) [reply cmd/kubeadm/app/util/etcd/etcd.go]: I thought about adding a flag, noResponseMap, to Client#GetClusterStatus so that empty map is returned when noResponseMap is true.

File: cmd/kubeadm/app/util/etcd/etcd.go
Comment (fabriziopandini) [inline cmd/kubeadm/app/util/etcd/etcd.go]: I don't think we should add a flag to the interface, because the only consumer that uses the return value is inside the etcd.go file.
what about removing the return value and renaming into CheckClusterHealth

File: cmd/kubeadm/app/util/etcd/etcd.go
Comment (fabriziopandini) [inline cmd/kubeadm/app/util/etcd/etcd.go]: AFAIK this function is used only by etcd.go, so IMO this method can go away from the interface...
Opinions?

File: cmd/kubeadm/app/util/etcd/etcd.go
Comment (fabriziopandini) [inline cmd/kubeadm/app/util/etcd/etcd.go]: If the previous comments are ok, this should change a little bit.
Probably we need a public method and a private method to accommodate all the need, but I leave to you to figure out the best way to implement this

**Review by fabriziopandini — CHANGES_REQUESTED** [review]
@tedyu thanks for stepping in and trying to cleanup this piece of code!
I added some comment because I think we should take this opportunity to clean-up the `ClusterInterrogator` interface (instead of adding a new flag)

File: cmd/kubeadm/app/util/etcd/etcd.go
Comment (tedyu) [reply cmd/kubeadm/app/util/etcd/etcd.go]: It is used in the following places:
```
cmd/kubeadm/app/phases/upgrade/compute_test.go:func (f fakeEtcdClient) ClusterAvailable() (bool, error) { return true, nil }
cmd/kubeadm/app/phases/upgrade/staticpods_test.go:func (c fakeTLSEtcdClient) ClusterAvailable() (bool, error) { return true, nil }
cmd/kubeadm/app/phases/upgrade/staticpods_test.go:func (c fakePodManifestEtcdClient) ClusterAvailable() (bool, error) { return true, nil }
```

**Comment by tedyu** [comment]
@fabriziopandini 
I have adopted your suggestion of func renaming.

Please take another look.

File: cmd/kubeadm/app/util/etcd/etcd.go
Comment (fabriziopandini) [reply cmd/kubeadm/app/util/etcd/etcd.go]: This confirms the fund isn't used in any logic, but  only in test when implementing fake objects,
Do you see some problems in removing this function from the interface/from fake objects?

File: cmd/kubeadm/app/util/etcd/etcd.go
Comment (fabriziopandini) [inline cmd/kubeadm/app/util/etcd/etcd.go]: Being this a private func, IMO we can make this return the map and let the consumers choose if to expose to the callers or not

**Review by fabriziopandini — CHANGES_REQUESTED** [review]
@tedyu thanks for the update!
Few more changes...

**Comment by tedyu** [comment]
@fabriziopandini 
Gentle ping.

File: cmd/kubeadm/app/util/etcd/etcd.go
Comment (neolit123) [inline cmd/kubeadm/app/util/etcd/etcd.go]: @tedyu
the removal of ClusterAvailable() seems good to me, but i don't see the benefits of the true/false flag to create or not create a map with responses. IMO it should be always created and returned, we are not chasing CPU cycles and nanoseconds here.

**Comment by tedyu** [comment]
@fabriziopandini @neolit123 
Your comment has been addressed.

**Comment by fabriziopandini** [comment]
@tedyu thanks for addressing all the comments!
/approve
I leave final lgtm to @neolit123 / other approvers

**Comment by neolit123** [comment]
thanks. will run a test on this and LGTM right after.
/test pull-kubernetes-e2e-kind

**Comment by neolit123** [comment]
looks like there are a lot of PRs in the tide queue, trying to make it in before code freeze.
given this is a cleanup it will have to wait after Code Thaw begins.
https://github.com/kubernetes/sig-release/tree/master/releases/release-1.16#timeline

we are entering code freeze for 1.16.

**Comment by tedyu** [comment]
@neolit123 
Can you give /lgtm ?

Thanks

**Comment by neolit123** [comment]
@tedyu 
we can merge this after the 10th when the code freeze is lifted.
currently only critical bug fixes are accepted.

**Pull request merged**
